CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+- Avoid merge commits in your PRs. If you want to pull in the latest changes from the `develop` branch, rebase on top of the `develop` branch instead of merging the `develop` branch into your feature branch.
+
+Also, ensure your commit messages match this format:
+
+    Short (50 characters or less) summary
+
+    After the 50 character summary and a blank line, you can include a body if necessary. Note that the 50 character summary does not end with any punctuation. Describe your changes in the imperative mood, e.g., "Add foo to bar", "Update foo component for bar", "Fix race condition for foo".
+
+    The body of the commit message can include:
+
+    * an explanation of the problem and what this change tries to solve.
+
+    * rationale behind the specific implementation
+
+    * alternatives considered and why they were discarded, if appropriate.
+
+    Please limit the line length in the body of a commit message to 80 characters or less.
+
+### Example Git Workflow
+
+Below is an example of how you can use git to create a feature branch. First, make sure you've created a fork of `aws/aws-cli`. Then you can run these commands:
+
+    # Clone the repo and set up the remotes.
+
+    $ git clone git@github.com:myusername/aws-cli.git
+    $ cd aws-cli
+    $ git remote add upstream https://github.com/aws/aws-cli.git
+    $ git fetch upstream
+    $ git merge upstream/develop
+
+    # Now to create a feature branch:
+    $ git checkout -b my-branch-name
+
+    # Now add your commits for your features.
+    $ git add path/to/my/files
+
+    # Make sure our commit message matches format described in the
+    # previous section.
+    $ git commit -m "Add support for foo"
+
+    # If we want to sync with the latest upstream changes before
+    # sending our pull request we can run:
+    $ git fetch upstream
+    $ git rebase upstream/develop
+
+    # When you're ready to send a PR, make sure you push your commits
+    # to your fork:
+    $ git push origin my-branch-name
+
+When you push to your remote, the output will contain a URL you can use to open a pull request.
+
+## CLI Development Version
+
+If you are interested in using the latest released version of the AWS CLI, please see the [Installation](README.md#installation) section in the README. This section is for anyone who wants to install the development version of the CLI. You might need to do this if:
+
+- You are developing a feature for the CLI and plan on submitting a Pull Request.
+- You want to test the latest changes of the CLI before they make it into an official release.
+
+The latest changes to the CLI are in the `develop` branch on github. This is the default branch when you clone the git repository.
+
+Additionally, there are several other packages that are developed in lockstep with the CLI. This includes:
+
+- [botocore](https://github.com/boto/botocore)
+- [jmespath](https://github.com/boto/jmespath)
```

Comment:
I think we can remove the references to jmespath here and later on. Botocore is really the only package the contributors may need to pull down as well.